tools/syz-cover/syz-cover.go

```diff
@@ -70,33 +70,26 @@ var (
 		"there are missing coverage callbacks")
 )
 
-func parseDates() (civil.Date, civil.Date) {
-	dateFrom, errDateFrom := civil.ParseDate(*flagDateFrom)
-	if errDateFrom != nil {
-		tool.Failf("failed to parse date from: %v", errDateFrom.Error())
-	}
-	dateTo, errDateTo := civil.ParseDate(*flagDateTo)
-	if errDateTo != nil {
-		tool.Failf("failed to parse date to: %v", errDateTo.Error())
-	}
-	return dateFrom, dateTo
-}
-
-func periodsFromDays(from, to civil.Date) []coveragedb.TimePeriod {
-	if to.Before(from) {
-		panic("toDay can't be less than fromDay")
-	}
-	res := []coveragedb.TimePeriod{{DateTo: from, Days: 1, Type: coveragedb.DayPeriod}}
-	for ; from.Before(to); from = from.AddDays(1) {
-		res = append(res, coveragedb.TimePeriod{DateTo: from, Days: 1, Type: coveragedb.DayPeriod})
+func dayPeriods(tp coveragedb.TimePeriod) []coveragedb.TimePeriod {
+	var res []coveragedb.TimePeriod
+	for i := 0; i < tp.Days; i++ {
+		res = append(res, coveragedb.TimePeriod{DateTo: tp.DateTo.AddDays(-i), Days: 1, Type: coveragedb.DayPeriod})
 	}
+	slices.Reverse(res)
```

Comment:
Nit: wouldn't it be easier to just loop from `tp.Days-1` to `0`?

Reply:
Agree.
This code was removed from the final version.